.github/workflows/pr-validate-fork.yaml

```diff
@@ -0,0 +1,105 @@
+# File inspired by https://github.com/Azure/azure-service-operator/blob/main/.github/workflows/pr-validation-fork.yml
+name: Validate PR (fork)
+on:
+  repository_dispatch:
+    types: [ok-to-test-command]
+
+env:
+  validate_job_name: validation-tests # must match env of same name in pr-validate
+
+jobs:
+  setup:
+    permissions:
+      checks: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: set-check-run-in-progress
+        uses: actions/github-script@v6
+        id: set-check-run-in-progress
+        env:
+          number: ${{ github.event.client_payload.pull_request.number }}
+          job: ${{ github.job }}
+          server_url: ${{ github.server_url }}
+          repo: ${{ github.repository }}
+          run_id: ${{ github.run_id }}
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const url = `${process.env.server_url}/${process.env.repo}/actions/runs/${process.env.run_id}`
+            const { data: pull } = await github.rest.pulls.get({
+              ...context.repo,
+              pull_number: process.env.number
+            });
+            const ref = pull.head.sha;
+            const { data: result } = await github.rest.checks.create({
+              ...context.repo,
+              name: process.env.validate_job_name,
+              head_sha: ref,
+              status: 'in_progress',
+              details_url: url,
+            });
+            return result;
+  e2e:
+    needs: setup
+    if:
+      github.event_name == 'repository_dispatch' &&
+      github.event.client_payload.slash_command.args.named.sha != '' &&
+      contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.args.named.sha)
```

Comment:
How does this completely stop risk of race condition?

Reply:
Verifying this though. That's how `if` should work because it's the same mechanism for cancelling jobs.

Reply:
I think that would probably be best, or if we can just make e2e run based off SHA if it is present

Reply:
Also whats the biggest thing we worry about with malicious code, with respect to github secrets it would have to be changes to actions right?

Reply:
patched things up so it's airtight here. The biggest worry with malicious code with respect to github secrets are two things
1. they can extract secrets if not they can get us to execute malicious code. In the context of this repo this isn't really a worry. The secrets we store aren't truly "secrets" because we use oidc which is a secretless authentication method. Only our github repo can authenticate with Azure through oidc. An attacker could see these secrets (which are more of ids like subscription id, service principal id, etc) but not utilize them in any way
2. They can execute malicious code on our infrastructure. Our e2e provisions aks clusters and runs tests on them. An injection of malicious code would allow an attacker to run that code on our cluster. Even further, they could get the kubeconfig or provision other infrastructure themselves. That's really what we must avoid here and we do with the fixes I added.